plugins/by-name/none-ls/prettierd.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, config, ... }:
+let
+  cfg = config.plugins.none-ls.sources.formatting.prettierd;
+  ts-ls-cfg = config.plugins.lsp.servers.ts_ls;
+in
+{
+  options.plugins.none-ls.sources.formatting.prettierd = {
+    disableTsServerFormatter = lib.mkOption {
+      type = with lib.types; nullOr bool;
+      description = ''
+        Disables the formatting capability of the `ts_ls` language server if it is enabled.
+      '';
+      default = null;
+      example = true;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    warnings = lib.optional ((cfg.disableTsServerFormatter == null) && ts-ls-cfg.enable) ''
+      You have enabled the `prettierd` formatter in none-ls.
+      You have also enabled the `ts_ls` language server which also brings a formatting feature.
+
+      - To disable the formatter built-in the `ts_ls` language server, set
+        `plugins.none-ls.sources.formatting.prettierd.disableTsServerFormatter` to `true`.
+      - Else, to silence this warning, explicitly set the option to `false`.
+    '';
```

Comment:
If this is the only reason we have `nullOr`, we can also use option priorities.

option-defaults have prio `1500` (aka `(lib.mkOptionDefault null).priority`), so we can check that a higher priority (well, lower number) is set for the option's _highest_ priority:

```nix
# Get access to this modules _options_ attrs
opt = options.plugins.none-ls.sources.formatting.prettierd;

# either
defaultPrio = (lib.mkOptionDefault null).priority;
# or
defaultPrio = 1500;

# Check isDefault instead of isNull
isDefault = opt.disableTsServerFormatter.highestPrio == defaultPrio;
```

_When designing the existing options, we probably didn't know about this technique..._